### PR TITLE
Add VS insertion PR tagger

### PR DESCRIPTION
**Pull request by allisonchou**
- Implements a PR tagger that creates an issue with the PRs that were inserted in a given VS build.
- Currently works with Roslyn and Razor but can likely be extended to other repos as well.
- After this is merged, we should be able to trigger runs automatically via AzDo pipelines.

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]
Nice work! Love that we can start re-using some of the infrastructure built up in this tool.

File: src/dotnet-roslyn-tools/Commands/PRTaggerCommand.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+internal static class PRTaggerCommand
+{
+    private static readonly PRTaggerCommandDefaultHandler s_prTaggerCommandHandler = new();
+
+    private static readonly Option<string> ProductName = new(new[] { "--product-name", "-p" }, "Name of product (e.g. 'roslyn' or 'razor')") { IsRequired = true };
+    private static readonly Option<string> VSBuild = new(new[] { "--build", "-b" }, "VS build number") { IsRequired = true };
+    private static readonly Option<string> CommitId = new(new[] { "--commit", "-c" }, "VS build commit") { IsRequired = true };
+    private static readonly Option<string> GitHubToken = new(new[] { "--github-token" }, "Token used to authenticate GitHub.") { IsRequired = true };
+    private static readonly Option<string> DevDivAzDOToken = new(new[] { "--devdiv-azdo-token" }, "Token used to authenticate to DevDiv Azure DevOps.") { IsRequired = true };
+    private static readonly Option<string> DncEngAzDOToken = new(new[] { "--dnceng-azdo-token" }, "Token used to authenticate to DncEng Azure DevOps.") { IsRequired = true };
```
Comment (JoeRobich) [inline src/dotnet-roslyn-tools/Commands/PRTaggerCommand.cs:20]: I think we should handle these tokens the same way `VSBranchInfo` does - https://github.com/dotnet/roslyn-tools/blob/main/src/dotnet-roslyn-tools/Commands/VSBranchInfoCommand.cs#L43. Add the token options from CommonOptions and then parse the setting from the `ParseResult`. Invalid/missing tokens could be checked up front or by simply allow the tagging to throw an exception with authorization issues occur.

File: src/dotnet-roslyn-tools/Commands/PRTaggerCommand.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+using System.CommandLine.Invocation;
+using System.CommandLine;
+using Microsoft.RoslynTools.Authentication;
+
+internal static class PRTaggerCommand
+{
+    private static readonly PRTaggerCommandDefaultHandler s_prTaggerCommandHandler = new();
+
+    private static readonly Option<string> ProductName = new(new[] { "--product-name", "-p" }, "Name of product (e.g. 'roslyn' or 'razor')") { IsRequired = true };
```
Comment (JoeRobich) [inline src/dotnet-roslyn-tools/Commands/PRTaggerCommand.cs:15]: We can use `.FromAmong(string[])` to ensure a valid product name. See https://docs.microsoft.com/en-us/dotnet/standard/commandline/define-commands#list-valid-argument-values

File: src/dotnet-roslyn-tools/PRTagger/PRTagger.cs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            new
+            {
+                title = $"[Automated] PRs inserted in VS build {vsBuildNumber}",
+                body = issueBody,
+                labels = new string[] { "vs-insertion" }
+            }));
+
+        if (!response.IsSuccessStatusCode)
+        {
+            logger.LogInformation($"Issue creation failed with status code: {response.StatusCode}");
```
Comment (JoeRobich) [inline src/dotnet-roslyn-tools/PRTagger/PRTagger.cs:198]: Maybe LogWarning or LogError.
```suggestion
            logger.LogError($"Issue creation failed with status code: {response.StatusCode}");
```

**Review by JoeRobich — APPROVED** [review]
Nice!

**Pull request merged**